### DISCUSSION TITLE
Force floating element to shift/flip when overlapping a specific element

### DISCUSSION THREAD

**Discussion by Ladvace** [floating-ui/floating-ui#2417] [Q&A] [answered]
I have a sidebar banner and I would like a floating element not to overlap it, is there a way to force a floating element not to overlap it,  shifting the floating element when the banner is close?

**Comment by atomiks** [comment, marked as answer]
This is a feature actively being tracked here: https://github.com/floating-ui/floating-ui/issues/1440

But there exists a couple of workarounds:

1. `flip` and `shift` accept a `padding` option which insets the boundary used to determine a collision. So for a sidebar on the left:

```js
flip({
  padding: {left: sidebarRect.right}, // distance of sidebar's right if the viewport is the boundary
})
```

2. Specify `boundary` to be a container that does not wrap the sidebar but the rest of the content instead

**Reply by Ladvace** [reply]
thanks, I'm gonna give it a try
